src/main/kotlin/com/exactpro/th2/check1/rule/sequence/NoMessageCheckTask.kt

```diff
@@ -0,0 +1,134 @@
+import com.exactpro.th2.check1.StreamContainer
+import com.exactpro.th2.check1.grpc.PreFilter
+import com.exactpro.th2.check1.rule.AbstractCheckTask
+import com.exactpro.th2.check1.rule.ComparisonContainer
+import com.exactpro.th2.check1.rule.MessageContainer
+import com.exactpro.th2.check1.rule.SailfishFilter
+import com.exactpro.th2.check1.rule.sequence.SequenceCheckRuleTask.Companion.PRE_FILTER_MESSAGE_NAME
+import com.exactpro.th2.check1.util.VerificationUtil
+import com.exactpro.th2.common.event.Event
+import com.exactpro.th2.common.grpc.EventBatch
+import com.exactpro.th2.common.grpc.EventID
+import com.exactpro.th2.common.grpc.MessageFilter
+import com.exactpro.th2.common.grpc.RootMessageFilter
+import com.exactpro.th2.common.message.toTreeTable
+import com.exactpro.th2.common.schema.message.MessageRouter
+import com.exactpro.th2.sailfish.utils.ProtoToIMessageConverter
+import com.google.protobuf.TextFormat
+import io.reactivex.Observable
+import java.time.Instant
+
+class NoMessageCheckTask(
+    description: String?,
+    startTime: Instant,
+    sessionKey: SessionKey,
+    messageTimeout: Long? = null,
+    timeout: Long,
+    maxEventBatchContentSize: Int,
+    protoPreFilter: PreFilter,
+    parentEventID: EventID,
+    messageStream: Observable<StreamContainer>,
+    eventBatchRouter: MessageRouter<EventBatch>
+    ) : AbstractCheckTask(description, messageTimeout, timeout, maxEventBatchContentSize, startTime, sessionKey, parentEventID, messageStream, eventBatchRouter) {
+
+    private val protoPreMessageFilter: RootMessageFilter = protoPreFilter.toRootMessageFilter()
+    private val messagePreFilter = SailfishFilter(
+        converter.fromProtoPreFilter(protoPreMessageFilter),
+        protoPreMessageFilter.toCompareSettings()
+    )
+
+    private val metadataPreFilter: SailfishFilter? = protoPreMessageFilter.metadataFilterOrNull()?.let {
+        SailfishFilter(
+            converter.fromMetadataFilter(it, VerificationUtil.METADATA_MESSAGE_NAME),
+            it.toComparisonSettings()
+        )
+    }
+
+    private lateinit var preFilterEvent: Event
+    private lateinit var resultEvent: Event
+
+    private var preFilterMessagesCounter: Int = 0
+    private var extraMessagesCounter: Int = 0
+
+    init {
+        rootEvent
+            .name("No message check $sessionKey")
+            .type("noMessageCheck")
+    }
+
+    override fun onStart() {
+        super.onStart()
+        preFilterEvent = Event.start()
+            .type("preFiltering")
+            .bodyData(protoPreMessageFilter.toTreeTable())
+        rootEvent.addSubEvent(preFilterEvent)
+        resultEvent = Event.start()
+            .type("noMessagesCheckResult")
+        rootEvent.addSubEvent(resultEvent)
+    }
+
+    override fun onNext(messageContainer: MessageContainer) {
+        LOGGER.debug(
+            "Received message with id: {}",
+            TextFormat.shortDebugString(messageContainer.protoMessage.metadata.id)
+        )
```

Comment:
You should wrap the call in `if` condition and call it only if the debug level is enabled